source/utils/index.ts

```diff
@@ -0,0 +1,155 @@
+import crypto from 'crypto';
+import https from 'https';
+
+/**
+ * Generate a UUID
+ */
+export function uuid(): string {
+  const uuidArray = new Uint8Array(16);
+  crypto.randomFillSync(uuidArray);
+  const hexStr = (b: number) => {
+    const s = b.toString(16);
+    return b < 0x10 ? '0' + s : s;
+  };
+  let result = '';
+  let i = 0;
+  for (; i < 4; i += 1) {
+    result += hexStr(uuidArray[i]);
+  }
+  result += '-';
+  for (; i < 6; i += 1) {
+    result += hexStr(uuidArray[i]);
+  }
+  result += '-';
+  for (; i < 8; i += 1) {
+    result += hexStr(uuidArray[i]);
+  }
+  result += '-';
+  for (; i < 10; i += 1) {
+    result += hexStr(uuidArray[i]);
+  }
+  result += '-';
+  for (; i < 16; i += 1) {
+    result += hexStr(uuidArray[i]);
+  }
+  return result;
+}
+
+/**
+ * Current unix time in seconds.
+ */
+export function currentTimestamp(millis = false): number {
+  return Math.floor(new Date().getTime() / (millis ? 1 : 1000));
+}
+
+type EndpointParameters = {
+  universe: string | undefined;
+  token: string | undefined;
+};
+/**
+ * Get universe and token from the endpoint.
+ */
+export function getEndpointParams(
+  endpoint: string,
+  token?: string,
+): EndpointParameters | undefined {
+  if (!endpoint) {
+    return undefined;
+  }
+
+  if (endpoint.indexOf('submit.backtrace.io') !== -1) {
+    const positionFilter = 'backtrace.io/';
+    const startPosition =
+      endpoint.indexOf('backtrace.io/') + positionFilter.length;
+    if (startPosition === -1) {
+      return undefined;
+    }
+    const indexOfTheEndOfTheUniverseName = endpoint.indexOf('/', startPosition);
+    if (indexOfTheEndOfTheUniverseName === -1) {
+      return undefined;
+    }
+    const universeName = endpoint.substring(
+      startPosition,
+      indexOfTheEndOfTheUniverseName,
+    );
+
+    if (!token) {
+      const lastSeparatorIndex = endpoint.lastIndexOf('/');
+      if (lastSeparatorIndex === indexOfTheEndOfTheUniverseName) {
+        return undefined;
+      }
+      token = endpoint.substring(
+        indexOfTheEndOfTheUniverseName + 1,
+        lastSeparatorIndex,
+      );
+      if (!token || token.length !== 64) {
+        return undefined;
+      }
+    }
+
+    return { universe: universeName, token };
+  }
+
+  const backtraceSubmissionUrl = new URL(endpoint).hostname;
+  const firstSeparatorIndex = backtraceSubmissionUrl.indexOf('.');
+  // unvalid submission URL
+  if (firstSeparatorIndex === -1) {
+    return undefined;
+  }
+  return {
+    token,
+    universe: backtraceSubmissionUrl.substring(0, firstSeparatorIndex),
+  };
+}
+
+/**
+ * Send POST request.
+ * @param url - string endpoint
+ * @param data - JSON Object.
+ */
+export async function post(
```

Comment:
here we don't have a similar problem that we had with backtrace-js. We can use axios and promises.

Reply:
Is this not a public package as well, where we should minimize dependencies? We don't get much functionality adding axios since I already rewrote this function haha.

Reply:
it's a node dependency - there is a different type of approach - server side vs client side. It doesn't matter how much time you spent on downloading our library in the web browser because you won't launch it in the web browser.

Reply:
Makes sense. But there is not enough reason to add axios as dependency here since we only make one post request.

Reply:
the axios dependency is already in this project.

Reply:
Oh, hah. Using it now.